base/src/garbage_collector.rs

```diff
@@ -0,0 +1,41 @@
+#![allow(clippy::unwrap_used)]
+
+use crate::model::Model;
+
+impl Model {
+    /// The garbage collector cleans up any lefover elements in the workbook that are no longer used.
+    /// These include:
+    /// - Shared strings that are no longer referenced by any cell
+    pub(crate) fn garbage_collector(&mut self) -> Result<(), String> {
+        let cell_values = self.get_all_cell_values().unwrap();
+        let mut new_shared_strings = self.shared_strings.clone();
```

Comment:
We have two copies of the shared strings :/. This one that you are cloning and `self.workbook.shared_strings`.

The second one is just the list of shared strings and mimics the Excel structure. The first one `self.shared_strings` is just an optimization device. It was added in 67e8b0ce145930b02048e2e5da4ed92384c9f859.

Before fixing this issue could you add a test that will fail without the issue.

Also we might not need to clone the shared string. What is there is nothing to clean up?

You might not need this, but Rust offers some utilities to work in situations similar to those. It's called Clone On Write.

You can see an example in:

https://github.com/ironcalc/IronCalc/blob/main/xlsx/src/export/escape.rs
https://doc.rust-lang.org/std/borrow/enum.Cow.html

```
                self.workbook.shared_strings.push(value.to_string());
                self.shared_strings.insert(value.to_string(), string_index);
```

Reply:
> Before fixing this issue could you add a test that will fail without the issue.

@nhatcher Will you elaborate on this?  What would this be testing exactly?

Reply:
Hey, sorry I missed this. The idea is to write a test that will fail right now without writing any code. Then you fix the issue and make sure that the test is passing (a _regression_ test_. We can discuss this over a call if it is unclear.

Reply:
I think I might be missing something, as both of these tests should cover a regression:

- https://github.com/ironcalc/IronCalc/pull/18/files#diff-d3ca2f9ce2aa275260078785b08b1515090994045364f03035823d9f3fa52528R136-R154
- https://github.com/ironcalc/IronCalc/pull/18/files#diff-d738915e77f990f230abcc69bac45acb3e3b7536a5fbc395dc076ad7f27f8bb6R4-R31